src/components/MainCategoryTap.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import { TEXT_DETAILS } from '@assets/constant/constant'
+
+//이 컴포넌트를 호출할때는 위로와 축하로 나눠질 상황에서 실행될 함수들을 꼭 props로 전달해주셔야 합니다.
+type MainCategory = {
+  onRecordCelebration: () => void
+  onRecordConsolation: () => void
+  currentRecordType: string
+}
+function MainCategoryTap({
+  onRecordCelebration,
+  onRecordConsolation,
+  currentRecordType,
+}: MainCategory) {
+  const { CELEBRATION, CONSOLATION } = TEXT_DETAILS
+
+  const typeConfig = {
+    active: 'text-primary-2 border-b-2 border-current pb-4',
+    inactive: 'text-gray-600',
+  }
+
+  const handleRecordTypeToCelebration = (): void => {
+    onRecordCelebration()
+  }
+
+  const handleRecordTypeToConsolation = (): void => {
+    onRecordConsolation()
+  }
```

Comment:
지금은 함수 2개, currentRecordtype 3개를 전달해주고 있는데 따로 내려주는거보단 currentRecordType, setRecordType 2개를 전달해줘서 구현하는게 더 좋을 것 같습니다. TEXT_DETAILS은 따로 임포트하구요. 그러면 `AddRecord.tsx` 에서도 따로 함수를 생성하지 않아도 되겠죠
```suggestion
  const handleRecordTypeToCelebration = (): void => {
    setRecordType(CELEBRATION);
  }

  const handleRecordTypeToConsolation = (): void => {
    setRecordType(CONSOLATION);
  }
```

Reply:
매우 공감합니다 바로적용할게영